Adding refactor for Movies project

xaindeneb opened the pull request:
Please review

xaindeneb committed:
Adding refactor for Movies project

xaindeneb committed:
Fix code style observations.

xaindeneb committed:
Merge remote-tracking branch 'upstream/develop' into develop

xaindeneb committed:
Fix code style observations

xaindeneb committed:
Fixing code style observation in Customer.java

xaindeneb committed:
Merge remote-tracking branch 'upstream/develop' into develop

xaindeneb committed:
Fix code style observations

carledriss commented on src/main/java/com/jalasoft/selenium/Ariel/movies/Children.java:
```diff
@@ -0,0 +1,26 @@
+package com.jalasoft.selenium.Ariel.movies;
```
packages should be in lowercase

xaindeneb replied:
Packages were renamed. They are in lower case now.

carledriss commented on src/main/java/com/jalasoft/selenium/Ariel/movies/Customer.java:
```diff
@@ -0,0 +1,91 @@
+
+class Customer {
+    private String name;
+    private Vector rentals = new Vector();
```
It is better to use a List instead of Vector.
```java
private List<Rental> rentals = new ArrayList<>();
```

xaindeneb replied:
Change to use a List instead of a Vector. Iteration was changed to use a for loop.

carledriss commented on src/main/java/com/jalasoft/selenium/Ariel/movies/Customer.java:
```diff
@@ -0,0 +1,91 @@
+     */
+    public String statement() {
+        Enumeration rentals = this.rentals.elements();
+        StringBuffer report = new StringBuffer();
```
It is better to use a StringBuilder instead of StringBuffer.

xaindeneb replied:
StringBuilder used instead of StringBuffer.

carledriss commented on src/main/java/com/jalasoft/selenium/Ariel/movies/Customer.java:
```diff
@@ -0,0 +1,91 @@
+            Rental each = (Rental) rentals.nextElement();
+            //show figures for this rental
+            String rentalFigure = "\t" + each.getMovie().getTitle()
+                    + "\t" + String.valueOf(each.getMovie().calculateAmount(each.getDaysRented())) + "\n";
```
String.valueOf is unnecessary here.
Remove it.

xaindeneb replied:
String.valueOf was removed.

carledriss commented on src/main/java/com/jalasoft/selenium/Ariel/movies/Customer.java:
```diff
@@ -0,0 +1,91 @@
+        while (rentals.hasMoreElements()) {
+            Rental each = (Rental) rentals.nextElement();
+            //show figures for this rental
+            String rentalFigure = "\t" + each.getMovie().getTitle()
```
It is not a good practice to concatenate string with +
take advantage of StringBuilder class

carledriss replied:
please use the append method

xaindeneb replied:
append method of StringBuilder was used to concatenate.

carledriss commented on src/main/java/com/jalasoft/selenium/Ariel/movies/Main.java:
```diff
@@ -0,0 +1,25 @@
+/**
+ * Created by Ariel Mattos on 3/6/2017.
+ */
+public final class Main {
```
now you can remove this class
since you have added unit tests.

carledriss replied:
:O 
It is missing unit test for movies classes

carledriss replied:
Please add unit tests

xaindeneb replied:
Main.java class removed. Unit tests are still pending.

carledriss commented:
@xaindeneb 
so far good job!

xaindeneb committed:
Changes according to code review suggestions.

carledriss commented on src/main/java/com/jalasoft/selenium/Ariel/movies/Customer.java:
```diff
@@ -0,0 +1,83 @@
+     */
+    public String statement() {
+        StringBuilder report = new StringBuilder();
+        report.append("Rental Record for ").append(getName()).append("\n");
```
you can use directly the attribute name here instead of calling to getName() method

carledriss commented on src/main/java/com/jalasoft/selenium/Ariel/movies/Customer.java:
```diff
@@ -0,0 +1,83 @@
+     * Calculate the total charges for all rented movies.
+     * @return              the total amount owed by the customer.
+     */
+    public double calculateTotalAmount() {
```
Take advantage of stream
please take a look at https://docs.oracle.com/javase/tutorial/collections/streams/reduction.html

carledriss commented on src/main/java/com/jalasoft/selenium/Ariel/movies/Customer.java:
```diff
@@ -0,0 +1,83 @@
+     * Calculate the total frequent renter points accumulated for all rented movies.
+     * @return              the total frequent renter points of the customer.
+     */
+    public int calculateFrequentRenterPoints() {
```
Take advantage of stream
please take a look at https://docs.oracle.com/javase/tutorial/collections/streams/reduction.html

carledriss commented on src/main/java/com/jalasoft/selenium/Ariel/movies/Main.java:
```diff
@@ -0,0 +1,25 @@
+/**
+ * Created by ariel Mattos on 3/6/2017.
+ */
+public final class Main {
```
Please remove this class

carledriss commented on src/main/java/com/jalasoft/selenium/Ariel/movies/Customer.java:
```diff
@@ -0,0 +1,83 @@
+        double totalAmount = 0;
+
+        for (Rental rental : rentals) {
+            totalAmount += rental.getMovie().calculateAmount(rental.getDaysRented());
```
you can improve this line of code.
Notice that you are using movie and daysRented here and you have that information in Rental class
so you can delegate this calculateAmount that to Rental class.

```java
totalAmount += rental.calculateAmount();
```

and in the Rental class add one method for calculateAmount.
```java
public double calculateAmount() {
    movie.calculateAmount(daysRented);
}
```

Apply the same to frequentRenterPoints

xaindeneb committed:
Modified according to the followin suggestions:
 - access the attribute name directly
 - use if streams in the functions to calculate totals
 - added calculate methods in the Rental class

carledriss commented:
LGTM!
It just missing unit test for refactor code.

xaindeneb committed:
Refactor calculation methods. Add unit test for Movie classes.

carledriss commented:
LGTM!
